examples/python/model-chat.py

```diff
@@ -31,67 +101,46 @@ def main(args):
+    prompt_tool_input = ""
+    guidance_input = ""
+    if args.guidance_type != "none":
+        guidance_type = args.guidance_type
+        if not args.guidance_info:
+            raise ValueError("Guidance information is required if guidance type is provided")
+        if guidance_type == "json_schema" or guidance_type == "lark_grammar":
+            tools_list = args.guidance_info
+            if guidance_type == "json_schema":
+                prompt_tool_input, guidance_input = get_json_grammar(tools_list)
+            elif guidance_type == "lark_grammar":
+                prompt_tool_input, guidance_input = get_lark_grammar(tools_list)
+        elif guidance_type == "regex":
+            guidance_input = args.guidance_info
         else:
-            raise ValueError(f"Chat Template for model type {model_type} is not known. Please provide chat template using --chat_template")
-
-    if args.verbose:
-        print("Model type is:", model_type)
-        print("Chat Template is:", args.chat_template)
+            raise ValueError("Guidance Type can only be [json_schema, regex, or lark_grammar]")
 
     params = og.GeneratorParams(model)
     params.set_search_options(**search_options)
+    if guidance_type:
+        params.set_guidance(guidance_type, guidance_input)
+        if args.verbose:
+            print("Guidance type is set to:", guidance_type)
+            print("Guidance input is:", guidance_input)
+
     generator = og.Generator(model, params)
     if args.verbose: print("Generator created")
-
-    # Set system prompt
-    if "<|" in args.system_prompt and "|>" in args.system_prompt:
-        # User-provided system template already has tags
-        system_prompt = args.system_prompt
+    if guidance_type == "json_schema" or guidance_type == "lark_grammar":
+        messages = f"""[{{"role": "system", "content": "{system_prompt}", "tools": "{prompt_tool_input}"}}]"""
     else:
-        if model_type.startswith('phi4'):
-            system_prompt = f"<|im_start|>system<|im_sep|>\n{args.system_prompt}<|im_end|>"
-        elif model_type.startswith('phi'): # For Phi2 and Phi3
-            system_prompt = f"<|system|>\n{args.system_prompt}<|end|>"
-        elif model_type.startswith("llama"):
-            system_prompt = f"<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n{args.system_prompt}<|eot_id|>"
-            print("Using System Prompt for LLAMA 3, if you are using LLAMA  2 please pass the argument --system_prompt '<s>[INST] <<SYS>>\\n{args.system_prompt}\\n<</SYS>>')")
-        elif model_type.startswith("qwen2"):
-            system_prompt = f"<|im_start|>system\n{args.system_prompt}<|im_end|>\n"
-        elif model_type == "gemma3_text":
-            system_prompt = f"{args.system_prompt}"
-        else:
-            system_prompt = args.system_prompt
-
-    system_tokens = tokenizer.encode(system_prompt)
-    generator.append_tokens(system_tokens)
-    system_prompt_length = len(system_tokens)
```

Comment:
`system_prompt_length = len(system_tokens)` should not be deleted, because we need to use it later:
if args.rewind:
            generator.rewind_to(system_prompt_length)